esp-hal/src/spi/master.rs

```diff
@@ -1579,6 +1579,12 @@ mod dma {
                 tx_buf,
             }
         }
+
+        /// Splits `SpiDmaBus` back into `SpiDma` to allow for manual
+        /// buffer control
+        pub fn split(self) -> (SpiDma<'d, M, T>, DmaRxBuf, DmaTxBuf) {
+            (self.spi_dma, self.rx_buf, self.tx_buf)
```

Comment:
There should probably be some kind of `flush()` call before this.

(Previously the type system would've caught this but it's no longer the case)

Reply:
Fixed